akka-docs/rst/scala/persistence-query.rst

```diff
@@ -0,0 +1,212 @@
+complex queries - this migration of data, possibly including some transformation along the way, can be easily implemented
+using akka persistence query as well - by pulling out of one Journal and storing into another one.
+
+.. warning::
+
+  This module is marked as **“experimental”** as of its introduction in Akka 2.4.0. We will continue to
+  improve this API based on our users’ feedback, which implies that while we try to keep incompatible
+  changes to a minimum the binary compatibility guarantee for maintenance releases does not apply to the
+  contents of the ``akka.persistence.query`` package.
+
+Design overview
+===============
+
+Akka persistence query is purposely designed to be a very loose API.
+This is in order to keep the provided APIs general enough for each journal implementation to be able to expose its best
+features, e.g. a SQL journal can use complex SQL queries or if a journal is able to subscribe to a live event stream
+this should also be possible to expose the same API - a typed stream of events.
+
+**Each read journal must explicitly document which types of queries it supports.**
+Refer to the your journal's plugins documentation for details on which queries and semantics it supports.
+
+While Akka Persistence Query does not provide actual implementations of ReadJournals, it defines a number of pre-defined
+query types for the most common query scenarios, that most journals are likely to implement (however they are not required to).
+
+Read Journals
+=============
+
+In order to issue queries one has to first obtain an instance of a ``ReadJournal``.
+Read journals are implemented as `Community plugins`_, each targeting a specific datastore (for example Cassandra or JDBC
+databases). For example, given a library that provides a ``akka.persistence.query.noop-read-journal`` obtaining the related
+journal is as simple as:
+
+.. includecode:: code/docs/persistence/query/PersistenceQueryDocSpec.scala#basic-usage
+
+Journal implementers are encouraged to put this identified in a variable known to the user, such that one can access it via
+``journalFor(NoopJournal.identifier)``, however this is not enforced.
+
+Read journal implementations are available as `Community plugins`_.
+
+
+Predefined queries
+------------------
+Akka persistence query comes with a number of ``Query`` objects built in and suggests journal implementors to implement
+them according to the semantics described below. It is important to notice that while these query types are very common
+a journal is not obliged to implement all of them - for example because in a given journal such query would be
+significantly inefficient.
+
+.. note::
+  Refer to the documentation of the ``ReadJournal`` plugin you are using for a specific list of supported query types.
+  For example, Journal plugins should document their stream completion strategies.
+
+The predefined queries are:
```

Comment:
not really, too much content inside (incl. code snippets), enumeration would be a bit weird here. I;ll reword the sentence instead maybe